src/session_concurrency/request_handler.cc

```diff
@@ -73,6 +73,10 @@ RequestHandler::SendRequestAndWaitForResponse(
 
   PayloadJsonUtils::UpdateHistoryAndAddToPayload(payload, chat_history);
 
+  auto& request_inputs{request_record.request_inputs_.emplace_back()};
+
+  RecordRequestInputs(payload, dataset_index, request_inputs);
```

Comment:
Request inputs are now recorded slightly earlier than they were before so that access to the other inputs (session ID, delay) are easier.